tests/mypy/modules/success.py

```diff
@@ -308,3 +309,18 @@ class Abstract(BaseModel):
 
 class Concrete(Abstract):
     class_id = 1
+
+
+def two_dim_shape_validator(v: Dict[str, Any]) -> Dict[str, Any]:
+    assert 'volume' not in v, 'shape is 2d, cannot have volume'
+    return v
+
+
+class Square(BaseModel):
+    width: float
+    height: float
+
+    @model_validator(mode='before')
+    @classmethod
+    def free_validator(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        return two_dim_shape_validator(values)
```

Comment:
```suggestion
    free_validator = model_validator(mode='before')(two_dim_shape_validator)
```
This is how the free validator should be defined I think, at least I think this is the thing that the issue wanted to pass type-checking.

It might make sense to keep a classmethod one as well, but for that I would use `def classmethod_validator(...` as the name.

Reply:
Yeah, the issue had `_normaliser = model_validator(mode='before')(normalise_known_invalids)` in the example of failure